numpy/lib/tests/test_nanfunctions.py

```diff
@@ -388,6 +410,94 @@ def test_empty(self):
             assert_equal(res, tgt)
 
 
+class TestNanFunctions_CumSumProd(TestCase, SharedNanFunctionsTestsMixin):
+
+    nanfuncs = [np.nancumsum, np.nancumprod]
+    stdfuncs = [np.cumsum, np.cumprod]
+
+    def test_allnans(self):
+        for f, tgt_value in zip(self.nanfuncs, [0, 1]):
+            with assert_no_warnings():
```

Comment:
add a comment that unlike other nan-functions, sum/prod/cumsum/cumprod don't warn on all nan input.